json-endpoints/jetpack/class.jetpack-json-api-cron-endpoint.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+
+class Jetpack_JSON_API_Cron_Endpoint extends Jetpack_JSON_API_Endpoint {
+	protected $needed_capabilities = 'manage_options';
+
+	protected function validate_call( $_blog_id, $capability, $check_manage_active = true ) {
+		parent::validate_call( $_blog_id, $capability, false );
+	}
+	protected function result() {
+
+	}
+
+}
+// GET /sites/%s/cron
+class Jetpack_JSON_API_Cron_Get_Endpoint extends Jetpack_JSON_API_Cron_Endpoint {
+	protected $needed_capabilities = 'manage_options';
+
+	protected function validate_call( $_blog_id, $capability, $check_manage_active = true ) {
+		parent::validate_call( $_blog_id, $capability, false );
+	}
+
+	protected function result() {
+		$cron = _get_cron_array();
+		return array(
+			'cron_array' => $cron,
+		    'current_timestamp' => time()
+		);
+	}
+}
+
+// POST /sites/%s/cron
+class Jetpack_JSON_API_Cron_Post_Endpoint extends Jetpack_JSON_API_Cron_Endpoint {
+	protected function result() {
+		define('DOING_CRON', true);
+
+		if ( false === $crons = _get_cron_array() )
+			return new WP_Error( 'no-cron-event', 'Current there are no cron events' );
+
+		$keys = array_keys( $crons );
+		$gmt_time = microtime( true );
+
+		if ( isset($keys[0]) && $keys[0] > $gmt_time )
+			return new WP_Error( 'no-cron-event', 'Current there are no cron events ready to be run' );
+
+
+		// The cron lock: a unix timestamp from when the cron was spawned.
+		$doing_cron_transient = $this->get_cron_lock();
+		if ( $doing_cron_transient && ( $doing_cron_transient + WP_CRON_LOCK_TIMEOUT > $gmt_time ) ) {
+			return new WP_Error( 'cron-is-locked', 'Current there is a cron already happening.' );
+		}
+
+		$doing_wp_cron = sprintf( '%.22F', microtime( true ) );
+		set_transient( 'doing_cron', $doing_wp_cron );
+		$processed_events = array();
+
+			foreach ( $crons as $timestamp => $cronhooks ) {
+			if ( $timestamp > $gmt_time )
+				break;
+
+			foreach ( $cronhooks as $hook => $keys ) {
```

Comment:
As discussed, we should probably be able to supply a comma-delimited set of cron jobs to run, and if that's set then we should check against it here so we only run the specified jobs.